src/Compilers/CSharp/Test/Semantic/Semantics/CollectionLiteralTests.cs

```diff
@@ -266,577 +423,788 @@ static void Main()
+                Diagnostic(ErrorCode.ERR_IdentifierExpected, "0").WithLocation(7, 13),
+                // (7,13): error CS1003: Syntax error, ']' expected
+                //         [3][0].GetHashCode();
+                Diagnostic(ErrorCode.ERR_SyntaxError, "0").WithArguments("]").WithLocation(7, 13),
+                // (7,14): error CS1002: ; expected
+                //         [3][0].GetHashCode();
+                Diagnostic(ErrorCode.ERR_SemicolonExpected, "]").WithLocation(7, 14),
+                // (7,14): error CS1513: } expected
+                //         [3][0].GetHashCode();
+                Diagnostic(ErrorCode.ERR_RbraceExpected, "]").WithLocation(7, 14),
+                // (7,16): error CS0120: An object reference is required for the non-static field, method, or property 'object.GetHashCode()'
+                //         [3][0].GetHashCode();
+                Diagnostic(ErrorCode.ERR_ObjectRequired, "GetHashCode").WithArguments("object.GetHashCode()").WithLocation(7, 16));
         }
 
         [Fact]
-        public void EnumType_02()
+        public void MemberAccess_03()
+        {
+            string source = """
+                class Program
+                {
+                    static void Main()
+                    {
+                        _ = [].GetHashCode();
+                        _ = []?.GetHashCode();
+                        _ = [][0].GetHashCode();
+                    }
+                }
+                """;
+            var comp = CreateCompilation(source);
+            comp.VerifyEmitDiagnostics(
+                // (5,16): error CS0117: 'collection literals' does not contain a definition for 'GetHashCode'
+                //         _ = [].GetHashCode();
+                Diagnostic(ErrorCode.ERR_NoSuchMember, "GetHashCode").WithArguments("collection literals", "GetHashCode").WithLocation(5, 16),
+                // (6,15): error CS0023: Operator '?' cannot be applied to operand of type 'collection literals'
+                //         _ = []?.GetHashCode();
+                Diagnostic(ErrorCode.ERR_BadUnaryOp, "?").WithArguments("?", "collection literals").WithLocation(6, 15),
+                // (7,13): error CS0021: Cannot apply indexing with [] to an expression of type 'collection literals'
+                //         _ = [][0].GetHashCode();
+                Diagnostic(ErrorCode.ERR_BadIndexLHS, "[][0]").WithArguments("collection literals").WithLocation(7, 13));
+        }
+
+        [Fact]
+        public void MemberAccess_04()
+        {
+            string source = """
+                class Program
+                {
+                    static void Main()
+                    {
+                        _ = [1].GetHashCode();
+                        _ = [2]?.GetHashCode();
+                        _ = [3][0].GetHashCode();
+                    }
+                }
+                """;
+            var comp = CreateCompilation(source);
+            comp.VerifyEmitDiagnostics(
```

Comment:
Are these errors dependent on the natural type decision?

Reply:
Yes, with a natural type, these expressions should be valid.